src/snippets/reference/nodejs/WithRule.js

```diff
@@ -43,10 +43,15 @@ const server = http.createServer(async function (req, res) {
   const decision = await getClient(userId).protect(req);
 
   if (decision.isDenied()) {
-    res.writeHead(429, { "Content-Type": "application/json" });
-    res.end(
-      JSON.stringify({ error: "Too Many Requests", reason: decision.reason }),
-    );
+    if (decision.reason.isRateLimit()) {
+      res.writeHead(429, { "Content-Type": "application/json" });
+      res.end(
+        JSON.stringify({ error: "Too Many Requests", reason: decision.reason }),
+      );
+    } else if (decision.reason.isBot()) {
```

Comment:
```suggestion
    } else {
```
This would just be an `else`, right?